roles/branch-validation/tasks/main.yml

```diff
@@ -155,7 +168,7 @@
     that: ping.user_input|bool
 
 - include: ha.yml
-  when: validate_ha_environment
+  when: groups['compute'] | intersect(groups['controller']) | length | bool
```

Comment:
This conditional here would prevent all of the HA testing from happening when on dedicated controller environments. That's not what we want.

We still want to test that HA works, it's just that we can't check if instances that were on controllers come back after a controller reboot. Those are the tasks that we need to add a condition to, and adjust the other tasks within that assume there being instances on controllers and computes.